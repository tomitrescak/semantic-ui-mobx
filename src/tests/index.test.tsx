import * as Form from "../index";

class Test extends Form.FormState {
  @Form.requiredField required: string = null;
  @Form.field ok = ''
  @Form.field(Form.emailValidator) badEmail = 'bad';
  @Form.field(Form.emailValidator) okEmail = 'ok@ok.com';
}

it('fails', () => {
  const test = new Test();
  expect(test.validate()).toMatchSnapshot();
})
