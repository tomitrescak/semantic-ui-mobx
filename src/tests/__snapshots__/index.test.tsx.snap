// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fails 1`] = `
"Field 'required': This field is required!
Field 'badEmail': Email has incorrect format!"
`;
